tests/unit/amo/components/TestReportUserAbuse.js

```diff
@@ -0,0 +1,142 @@
+  }
+
+  it('renders a disabled button if no user exists', () => {
+    const root = renderShallow({ user: null });
+
+    expect(root.find('.ReportUserAbuse')).toHaveLength(1);
+    expect(root.find('.ReportUserAbuse-show-more'))
+      .toHaveProp('disabled', true);
+  });
+
+  it('renders a button to report an user', () => {
+    const root = renderShallow();
+
+    expect(root.find('.ReportUserAbuse')).toHaveLength(1);
+    expect(root.find('.ReportUserAbuse-show-more').prop('children'))
+      .toEqual('Report this user for abuse');
+  });
+
+  it('shows the preview content when first rendered', () => {
+    const root = renderShallow();
+
+    expect(root.find('.ReportUserAbuse--is-expanded')).toHaveLength(0);
+  });
+
+  it('shows more content when showReportUI is called', () => {
+    const { store } = dispatchClientMetadata();
+    const dispatchSpy = sinon.spy(store, 'dispatch');
+    const user = createUserAccountResponse();
+    const root = renderShallow({ store, user });
+
+    root.instance().showReportUI();
+
+    sinon.assert.calledWith(dispatchSpy, showUserAbuseReportUI({ user }));
+  });
+
+  it('hides more content when hideReportUI is called', () => {
+    const { store } = dispatchClientMetadata();
+    const dispatchSpy = sinon.spy(store, 'dispatch');
+    const user = createUserAccountResponse();
+    const root = renderShallow({ store, user });
+
+    root.instance().showReportUI();
+    dispatchSpy.reset();
+
+    root.instance().hideReportUI();
+
+    sinon.assert.calledWith(dispatchSpy, hideUserAbuseReportUI({ user }));
+  });
+
+  it('dispatches the send abuse report action', () => {
+    const { store } = dispatchClientMetadata();
+    const dispatchSpy = sinon.spy(store, 'dispatch');
+    const user = createUserAccountResponse();
+    const root = renderShallow({ store, user });
+
+    root.instance().sendReport({ text: 'This user is funny' });
+
+    sinon.assert.calledWith(dispatchSpy, sendUserAbuseReport({
+      errorHandlerId: root.instance().props.errorHandler.id,
+      message: 'This user is funny',
+      user,
+    }));
+  });
+
+  it('shows a success message and hides the button if report was sent', () => {
+    const user = createUserAccountResponse();
+    const { store } = dispatchClientMetadata();
+    const abuseResponse = createFakeUserAbuseReport({
+      message: 'Seriously, where is my money?!',
+      user,
+    });
+    store.dispatch(loadUserAbuseReport(abuseResponse));
+    const root = renderShallow({ store, user });
+
+    expect(root.find('.ReportUserAbuse--report-sent')).toHaveLength(1);
+    expect(root.find('.ReportUserAbuse-form')).toHaveLength(0);
+    expect(root.find('.ReportUserAbuse-show-more')).toHaveLength(0);
+    expect(root.find('button.ReportUserAbuse-send-report')).toHaveLength(0);
+  });
+
+  it('renders an error if one exists', () => {
+    const errorHandler = createStubErrorHandler();
+    const { store } = dispatchClientMetadata();
+    store.dispatch(setError({
```

Comment:
It may be a little safer to use the `ErrorHandler` interface instead of calling the action directly.

```js
    const errorHandler = new ErrorHandler({
      id: 'some-error-handler-id',
      dispatch: store.dispatch,
    });
    errorHandler.handle(new Error('something bad'));
```

We should probably cleanup all the varying examples of how to do this.